src/components/Activities/Activity1/Game1/components/CipherGameRound.tsx

```diff
@@ -56,19 +56,31 @@ function CipherGameRound(props : CipherGameRoundProps): JSX.Element {
   const [slideIdx, setSlideIdx] = useState(0);
   const [happiness, setHappiness] = useState(0);
   const [clickDisabled, setClickDisabled] = useState(false);
+  const [hoverIncorrect, setHoverIncorrect] = useState(false);
+  const [roundNum, setRoundNum] = useState(0);
+
+  useEffect(()=>{
+    if (roundNum >   THRESHOLD_TO_HELP_PER_GAME ) {
+      hoverIncorrect ? handleAlienState(ALIEN_STATE.SAD) : handleAlienState(ALIEN_STATE.BASE);
+    }
+  }, [hoverIncorrect]);
 
   // -----ALIEN HANDLERS------
   const [alienState, setAlienState] = useState(ALIEN_STATE.BASE);
   const alienTimeout = useRef<NodeJS.Timeout | undefined>(undefined);
 
-  const handleAlienState = (state: ALIEN_STATE, cb?: () => void) => {
-    alienTimeout.current && clearTimeout(alienTimeout.current);
+  const handleAlienState = (state: ALIEN_STATE, timeout?: boolean, cb?: () => void) => {
     setAlienState(state);
-    alienTimeout.current = setTimeout(() => {
-      alienTimeout.current = undefined;
-      setAlienState(ALIEN_STATE.BASE);
-      cb && cb();
-    }, 1000);
+    if(timeout === true) {
+      alienTimeout.current && clearTimeout(alienTimeout.current);
+      alienTimeout.current = setTimeout(() => {
+        alienTimeout.current = undefined;
+        setAlienState(ALIEN_STATE.BASE);
+        cb && cb();
+      }, 1000);
+    } else {
+      setAlienState(state);
+    }
```

Comment:
your running setAlienState twice in the case where timeout is false


```suggestion
    if(timeout === false) return;
    alienTimeout.current && clearTimeout(alienTimeout.current);
    alienTimeout.current = setTimeout(() => {
      alienTimeout.current = undefined;
      setAlienState(ALIEN_STATE.BASE);
      cb && cb();
    }, 1000);
```

Reply:
also.. what if we did the check to see if cb is defined?

seems to me that having a callback is deterministic if we should run a timeout?

Reply:
sure sounds good! we could merge this part of this file with [this other duplicated file](https://github.com/uclaacm/Playnet/blob/cab134e886701690aabcc789f8f357294344397b/src/components/Activities/Activity1/Game2/components/GameSlide.tsx) when we refactor sometime!

Reply:
yeah we could make it its own hook if we want to!

Reply:
ooo thats cool ive never done hehe, will try sometime!